qualtran/_infra/single_qubit_controlled.py

```diff
@@ -0,0 +1,84 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import abc
+from typing import Iterable, Optional, Sequence, Tuple, TYPE_CHECKING
+
+import attrs
+
+from qualtran._infra.bloq import Bloq
+from qualtran._infra.controlled import CtrlSpec
+from qualtran._infra.registers import Register
+
+if TYPE_CHECKING:
+    from qualtran import AddControlledT, BloqBuilder, SoquetT
+
+
+class SpecializedSingleQubitControlledGate(Bloq):
```

Comment:
Should we replace the `Gate` suffix with a `Bloq` suffix? cc @mpharrigan for suggestions on naming the API

Reply:
Although Bloq(Bloq) is a bit redundant, SpecializedSingleQubitControlledMixin?

Reply:
I chose -Mixin